src/routes/safe/container/index.jsx

```diff
@@ -34,13 +34,23 @@ class SafeView extends React.Component<Props, State> {
 
   componentDidMount() {
     const {
-      fetchSafe, activeTokens, safeUrl, fetchTokenBalances, fetchTokens, fetchTransactions, fetchCurrencyValues, loadAddressBook,
+      fetchSafe,
+      activeTokens,
+      safeUrl,
+      fetchTokenBalances,
+      fetchTokens,
+      fetchTransactions,
+      fetchCurrencyValues,
+      loadAddressBook,
+      loadCurrentSession,
+      addViewedSafe,
     } = this.props
 
-    fetchSafe(safeUrl).then(() => {
+    fetchSafe(safeUrl)
       // The safe needs to be loaded before fetching the transactions
-      fetchTransactions(safeUrl)
-    })
+      .then(() => fetchTransactions(safeUrl))
+      .then(() => loadCurrentSession())
```

Comment:
why do we wait for safe to be fetched before loading the session? shouldn't we load the session as early as possible? (e.g index.js or Root component)

Reply:
I let everything as close as possible to avoid side effects, and completely forgot about it.

Thanks for the catch